ui/components/src/LineChart/LineChart.tsx

```diff
@@ -133,18 +145,89 @@ export function LineChart({ height, data, yAxis, unit, grid, legend, onDataZoom,
 
     const rangeMs = data.rangeMs ?? getDateRange(data.xAxis);
 
+    let series = data.timeSeries;
+
+    const startTime = data.xAxis[0];
+    const endTime = data.xAxis[data.xAxis.length - 1];
+
+    if (annotations.length > 0) {
```

Comment:
Yeah I need to just move the thresholds mapping into the chart this way too

Reply:
Yeah basically this component should do most of the work. The dashboard level component should just plugin in the correct context